extensions/ql-vscode/src/run-remote-query.ts

```diff
@@ -0,0 +1,57 @@
+import { Uri } from 'vscode';
+import * as yaml from 'js-yaml';
+import * as fs from 'fs-extra';
+import { showAndLogErrorMessage, showAndLogInformationMessage } from './helpers';
+import { Credentials } from './authentication';
+
+interface Config {
+  repositories: string[];
+  ref?: string;
+  language: string;
+}
+
+export default async function runRemoteQuery(credentials: Credentials, uri?: Uri) {
+  if (!uri || !uri.fsPath.endsWith('.ql')) {
+    return;
+  }
+
+  const octokit = await credentials.getOctokit();
+  const token = await credentials.getToken();
+
+  const queryFile = uri.fsPath;
+  const query = await fs.readFile(queryFile, 'utf8');
+
+  const repositoriesFile = queryFile.substring(0, queryFile.length - '.ql'.length) + '.repositories';
+  if (!(await fs.pathExists(repositoriesFile))) {
+    void showAndLogErrorMessage(`Missing file: '${repositoriesFile}' to specify the repositories to run against.`);
+    return;
+  }
+
+  const config = yaml.safeLoad(await fs.readFile(repositoriesFile, 'utf8')) as Config;
+
+  const ref = config.ref || 'main';
+  const language = config.language;
+  const repositories = JSON.stringify(config.repositories);
+  // const queryRunGuid = `${path.basename(queryFile)}-${Date.now()}`;
+  const owner = 'dsp-testing';
+  const repo = 'qc-controller';
+
+  try {
+    await octokit.rest.actions.createWorkflowDispatch({
+      owner: owner,
+      repo: repo,
+      workflow_id: 'codeql-query.yml',
+      ref: ref,
+      inputs: {
+        language,
+        repositories,
+        query,
+        token: token
```

Comment:
Since the field name is the same as the variable you're assigning to the field, this could also just be `token`, and then it matches the style of the other lines.
```suggestion
        token
```

Or change the others to `language: language,` etc... Just a bit odd to have both styles mixed together in one object.